### ISSUE TITLE
Incoming call notification on Pebble shows duration instead of phone number

### ISSUE THREAD

**Issue by bhaskarmurthy** [Freeyourgadget/Gadgetbridge#1193]
#### Before opening an issue please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
Since upgrade to Android P, incoming call notifications show duration instead of phone number.

Debugging `DeviceCommunicationService.java:L440` shows this:
```
callSpec = {CallSpec@12245} 
 command = 2
 name = "Test"
 number = "Test"
```

![screenshot_20180807-101325](https://user-images.githubusercontent.com/814369/43813435-b1a167d2-9a93-11e8-9144-5b706bc8660c.png)

[gadgetbridge.log](https://github.com/Freeyourgadget/Gadgetbridge/files/2268759/gadgetbridge.log)

#### Your wearable device is:
Device: `Pebble Time`
FW: `v4.3`
HW: `snowy_dvt`
*Please specify model and firmware version if possible*

#### Your android version is:
9

#### Your Gadgetbridge version is:
0.28.1

*New issues about already solved/documented topics could be closed without further comments. Same for too generic or incomplete reports.*

**Comment by bhaskarmurthy** [comment]
Only on Android 9 - Pie
Works on my other device that's running Android 8 - Oreo

**Comment by skweee** [comment]
Your other device is also a Pebble Time or Pebble 2?

**Comment by skweee** [comment]
I can confirm that this issue is not related to Android 9. I can reproduce this issue with my Pebble 2 HR (silk, FW 4.3) using Android 6.0.1 as well as Android 9.

Is there any protocol documentation for the newer Pebble models? I'd be willing to look into this, but I would need some more information on how Pebble expects incoming call notifications.

**Comment by cpfeiffer** [comment]
Does this also happen with your fix from #1252?

**Comment by skweee** [comment]
Yes, this is unrelated to #1252. This issue can be reproduced from the Debug screen in GB with the "Incoming Call" button, and as @bhaskarmurthy wrote, the `CallSpec` object looks exactly like it should be.

I already went through and debugged the code from the Debug screen down to the Bluetooth stack. My guess is that there is a problem either with the BTLE devices or with the newer Pebble firmwares, as some Pebble devices don't seem to be affected.

Unfortunately I don't have a reference for the message protocol for these devices. Do you have any pointer for me where I could find one (apart from sniffing the bluetooth traffic from the official app)?

**Comment by skweee** [comment]
Just to make the issue description a little clearer (if that helps): for incoming calls (`CallSpec.CALL_INCOMING`) the `CallSpec.number` is correctly populated. I followed it down to `PebbleGATTServer.sendDataToPebble(byte[] data)`, where you can find it in the byte array (with the correct length and content). However, the result is the screenshot @bhaskarmurthy posted, where on the very bottom you should be able to read "Test" instead of "426026:13:22".

**Comment by ashimokawa** [comment]
Does this also happen after rebooting your pebble?

**Comment by skweee** [comment]
Thank you, @ashimokawa. I'm feeling slightly stupid now, but rebooting the pebble really did change something. Now the duration at the bottom is gone:
![screenshot_20180906-105739](https://user-images.githubusercontent.com/7595525/45185066-3b1d9d80-b229-11e8-8b8b-de921b58a38f.png)
I was expecting the `number` information to be displayed instead. Is this the correct / intended behavior?

This is the screenshot of the "Find lost device" functionality. From `PebbleProtocol:587` I would have expected a line "Where are you?" below it. 
![screenshot_20180906-105803](https://user-images.githubusercontent.com/7595525/45185067-3b1d9d80-b229-11e8-95d0-4b9dea3b90a1.png)

**Comment by bhaskarmurthy** [comment]
FWIW: I rebooted my phone and pebble both, and doesn't fix the issue
